src/app/application-dialog/application-dialog.component.ts

```diff
@@ -0,0 +1,118 @@
+  euriborValues: Euribor[] = euriborValuesConst;
+  paymentScheduleTypes: string[] = ['Annuity', 'Linear'];
+  obligationFields = [
+    {label: 'Mortgage Loans', controlName: 'mortgageLoans'},
+    {label: 'Consumer Loans', controlName: 'consumerLoans'},
+    {label: 'Leasing Amount', controlName: 'leasingAmount'},
+    {label: 'Credit Card Limit', controlName: 'creditCardLimit'},
+  ];
+
+  ngOnInit() {
+    this.apiService.getConstants().subscribe((constants) => {
+      this.constants = constants;
+      this.minLoanTerm = constants.minLoanTerm;
+      this.maxLoanTerm = constants.maxLoanTerm;
+      this.loanAmountPercentage = constants.loanAmountPercentage;
+      this.maxKids = constants.maxKids;
+      this.maxMonthlyObligationsPercentage = constants.maxMonthlyObligationsPercentage;
+    });
+  }
+
+  isLinear = true;
+
+  loanDetailsForm = formBuilder.group({
+    realEstatePrice: [this.applicationData.realEstatePrice, Validators.required],
+    downPayment: [this.applicationData.downPayment, Validators.required],
+    loanAmount: [this.applicationData.loanAmount, Validators.required],
+    loanTerm: [this.applicationData.loanTerm, Validators.required],
+    euribor: [this.applicationData.euribor, [Validators.required]],
+    paymentScheduleType: [this.paymentScheduleTypes[0] as string, [Validators.required]],
+  });
+  incomeDetailsForm = formBuilder.group({
+    applicants: [this.applicationData.applicants, Validators.required],
+    amountOfKids: [this.applicationData.amountOfKids, Validators.required],
+    income: [this.applicationData.income, Validators.required],
+    obligations: [this.applicationData.obligations, Validators.required],
+    mortgageLoans: [this.applicationData.mortgageLoans],
+    consumerLoans: [this.applicationData.consumerLoans],
+    leasingAmount: [this.applicationData.leasingAmount],
+    creditCardLimit: [this.applicationData.creditCardLimit],
+    monthlyPayment: [this.applicationData.monthlyPayment],
+  });
+  personalDetailsForm = formBuilder.group({
+    firstName: [this.applicationData.firstName, Validators.required],
+    lastName: [this.applicationData.lastName, Validators.required],
+    personalNumber: [this.applicationData.personalNumber, Validators.required],
+    email: [this.applicationData.email, Validators.required],
+    phoneNumber: [this.applicationData.phoneNumber, Validators.required],
+    address: [this.applicationData.address, Validators.required]
+  });
+
+  get obligations() {
+    return this.incomeDetailsForm.get('obligations');
+  }
+
+  onCancelClick(): void {
+    this.dialogRef.close();
+  }
+
+  onDoneClick(): void {
+    //form validation and post to backend
+    this.saveLoanDetails();
+    console.log(`The dialog was closed with result: ${this.applicationData.realEstatePrice}`);
+  }
+
+  saveLoanDetails(): void {
+    this.applicationData.realEstatePrice = this.loanDetailsForm.value.realEstatePrice;
+//update other fields
+  }
+
+  maxApplicants(): number[] {
```

Comment:
Can we have this variable generated after fetching constants and not to be generated every time template check happens?